### PR TITLE
Make L4 NetLB Cpntroller Healthcheck return error

### PR DESCRIPTION
**Pull request by cezarygerard**
This is required to correctly track availability SLO when L4 NetLB Controller get's stuck

### CODE REVIEW

File: pkg/l4lb/l4netlbcontroller.go [hunk 1]
```diff
@@ -263,8 +262,7 @@ func (lc *L4NetLBController) checkHealth() error {
 	// This indicates that the controller was stuck handling a previous update, or sync function did not get invoked.
 	syncTimeLatest := lastEnqueueTime.Add(enqueueToSyncDelayThreshold)
 	if lastSyncTime.After(syncTimeLatest) {
-		msg := fmt.Sprintf("L4 External LoadBalancer Sync happened at time %v - %v after enqueue time, threshold is %v", lastSyncTime, lastSyncTime.Sub(lastEnqueueTime), enqueueToSyncDelayThreshold)
-		klog.Error(msg)
+		return fmt.Errorf("L4 External LoadBalancer Sync happened at time %v - %v after enqueue time, threshold is %v", lastSyncTime, lastSyncTime.Sub(lastEnqueueTime), enqueueToSyncDelayThreshold)
```
[the remark this answers is not in the document]
Comment (cezarygerard) [reply pkg/l4lb/l4netlbcontroller.go]: fixed.

**Comment by swetharepakula** [comment]
/lgtm
I will let @freehan review once and give final approval.

File: pkg/l4lb/l4controller.go [hunk 2]
```diff
@@ -144,8 +144,9 @@ func (l4c *L4Controller) checkHealth() error {
 	syncTimeLatest := lastEnqueueTime.Add(enqueueToSyncDelayThreshold)
 	if lastSyncTime.After(syncTimeLatest) {
 		msg := fmt.Sprintf("L4 ILB Sync happened at time %v - %v after enqueue time, threshold is %v", lastSyncTime, lastSyncTime.Sub(lastEnqueueTime), enqueueToSyncDelayThreshold)
+		// Log here, context/http handler do no log the error.
 		klog.Error(msg)
-		// TODO return error here
+		return fmt.Errorf(msg)
```
Comment (bowei) [inline pkg/l4lb/l4controller.go:149]: Error not Errorf

File: pkg/l4lb/l4netlbcontroller.go [hunk 3]
```diff
@@ -264,7 +263,9 @@ func (lc *L4NetLBController) checkHealth() error {
 	syncTimeLatest := lastEnqueueTime.Add(enqueueToSyncDelayThreshold)
 	if lastSyncTime.After(syncTimeLatest) {
 		msg := fmt.Sprintf("L4 External LoadBalancer Sync happened at time %v - %v after enqueue time, threshold is %v", lastSyncTime, lastSyncTime.Sub(lastEnqueueTime), enqueueToSyncDelayThreshold)
+		// Log here, context/http handler do no log the error.
 		klog.Error(msg)
+		return fmt.Errorf(msg)
```
Comment (bowei) [inline pkg/l4lb/l4netlbcontroller.go:268]: error, not Errorf

**Comment by bowei** [comment]
Are we sure we want to immediately turn this healthcheck on vs having a flag and seeing how the metric behaves? If this is too sensitive when it rolls out to production, the only thing we can do at that point is to rollback.

**Pull request merged**